### DISCUSSION TITLE
Adding labels to generated CRD

### DISCUSSION THREAD

**Discussion by katheris** [fabric8io/kubernetes-client#4475] [Q&A]
Hi, I am using fabric8 to generate a CRD. Is there any way to tell it to add some specific labels to the CRD it generates?

**Comment by csviri** [comment]
Hi, just a hist this might be also solvable in a different way, like using `kustomize` to add the labels, as a build step.

**Reply by katheris** [reply]
I agree that I could do it as a separate step. However, I wondered if I could provide it as part of the Java class for my CRD, alongside things like the group, version and shortnames. Or perhaps as an argument in maven.

**Reply by csviri** [reply]
Yep, maybe an additional annotation? 
(only smell with that is if there are multiple versions of the CRD, that is represented by more classes, then on which class to add the annotation, or both?.)

**Reply by katheris** [reply]
I guess the tricky thing is if you have labels that aren't always needed for the CRD, but you want in your particular build for your environment. In which case having it as a build step makes more sense. But then it does feel a bit hacky to be editing the generated file later. Especially with the "# Generated by Fabric8 CRDGenerator, manual edits might get overwritten!" warning at the top

**Comment by andreaTP** [comment]
It does make total sense to me to have first-class support for this.
Here you have a tentative implementation of the feature:
https://github.com/fabric8io/kubernetes-client/pull/4476
cc. @metacosm
